cmd/pulumi-test-language/tests.go

```diff
@@ -984,6 +984,93 @@ var languageTests = map[string]languageTest{
 			},
 		},
 	},
+	// This test checks how SDKs propagate properties marked as secret to the provider Configure on the gRPC level.
+	"l2-provider-grpc-config-schema-secret": {
+		providers: []plugin.Provider{&providers.ConfigGrpcProvider{}},
+		runs: []testRun{
+			{
+				assert: func(l *L,
+					projectDirectory string, err error,
+					snap *deploy.Snapshot, changes display.ResourceChanges,
+				) {
+					g := &grpcTestContext{l: l, s: snap}
+
+					// Verify the CheckConfig request received by the provider.
+					r := g.CheckConfigReq("config")
+
+					// CheckConfig request gets the secrets in the plain. This is suspect, probably
+					// has to do with secret negotiation happening later in the gRPC provider cycle.
+					assert.Equal(l, "SECRET",
+						r.News.Fields["secretString1"].AsInterface(), "secretString1")
+
+					assertEqualOrJSONEncoded(l, float64(16),
+						r.News.Fields["secretInt1"].AsInterface(), "secretInt1")
+
+					assertEqualOrJSONEncoded(l, float64(123456.7890),
+						r.News.Fields["secretNum1"].AsInterface(), "secretNum1")
+
+					assertEqualOrJSONEncoded(l, true,
+						r.News.Fields["secretBool1"].AsInterface(), "secretBool1")
+
+					assertEqualOrJSONEncoded(l, []any{"SECRET", "SECRET2"},
+						r.News.Fields["listSecretString1"].AsInterface(), "listSecretString1")
+
+					assertEqualOrJSONEncoded(l, map[string]any{"key1": "SECRET", "key2": "SECRET2"},
+						r.News.Fields["mapSecretString1"].AsInterface(), "mapSecretString1")
+
+					// TODO Languages do not agree on the object property casing sent to CheckConfig.
```

Comment:
Is this a known bug in Python SDK? This seems a little concerning.

Reply:
I think it's about how codegen generates the python code from the PCL.  I'm not super familiar with codegen though, so maybe @Zaid-Ajaj has more thoughts on this.  If we leave this as TODO it should have an issue attached to it, though I'm not sure this should be a todo.

Reply:
I've filed a bug. It's encouraging if it's just a programgen bug but one can squint at it and decide it's a product issue even without PCL/programgen. I'm not too concerned but I'm leaving the x-linked TODO.